Source/WebCore/css/parser/CSSParserFastPaths.cpp

```diff
@@ -481,6 +511,106 @@ static std::optional<SRGBA<uint8_t>> parseHexColorInternal(std::span<const Chara
     return finishParsingHexColor(value, characters.size());
 }
 
+template<typename CharacterType> static std::optional<SRGBA<uint8_t>> parseLegacyHSL(std::span<const CharacterType> characters)
+{
+    // Commas only exist in the legacy syntax.
+    size_t delimiter = find({ characters.data(), characters.data() + characters.size() }, ',');
+    if (delimiter == notFound)
+        return std::nullopt;
+
+    auto skipWhitespace = [](std::span<const CharacterType>& characters) ALWAYS_INLINE_LAMBDA {
+        while (!characters.empty() && isCSSSpace(characters.front()))
+            characters = characters.subspan(1);
+    };
+
+    auto parsePercentageWithOptionalLeadingWhitespace = [&](std::span<const CharacterType>& characters) -> std::optional<double> {
+        skipWhitespace(characters);
+
+        double value = 0;
+        size_t numCharactersParsed = parseDouble(characters, '%', value);
+        if (!numCharactersParsed)
+            return std::nullopt;
+
+        characters = characters.subspan(numCharactersParsed);
+        if (characters.empty() || characters.front() != '%')
+            return std::nullopt;
+
+        characters = characters.subspan(1); // Skip the '%'.
+        return value;
+    };
+
+    auto parseAlpha = [&](std::span<const CharacterType>& characters) -> std::optional<double> {
```

Comment:
I like defining the lambda right before their first use, makes code generally more sequential and easier to read